sdk/python/feast/infra/registry/proto_registry_utils.py

```diff
@@ -23,6 +24,25 @@
 from feast.stream_feature_view import StreamFeatureView
 
 
+def registry_proto_cache(func):
+    cache_key = None
+    cache_value = None
```

Comment:
I'm not sure I understand this.. If you use the same annotation for multiple methods then the cache value needs have different shapes, such as `List[FeatureService]` or `List[FeatureView]` right? Is that going to work as expected here? Can we add tests?

Reply:
The context of the wrapper decorator is created whenever different argument functions (e.g list_feature_views, list_feature_services..) are called. 
Therefore, when calling list_feature_views and when calling list_feature_services, different results are returned.
The operation of this cache is similar to Python's functools.lru_cache, but since the registryProto object is not hashable, it is implemented to hash using the object's id.